domain/user/state/state.go

```diff
@@ -175,20 +175,21 @@ WHERE name = $M.name AND removed = false
 	return usr, nil
 }
 
-// GetAllUsers will retrieve all users from the database where the user is
-// active and has not been removed. If no users exist an empty slice will be
-// returned.
-func (st *State) GetAllUsers(ctx context.Context) ([]user.User, error) {
+// GetAllUsersWithAuthInfo will retrieve all users with authentication information
+// (last login, disabled) from the database. If no users exist an empty slice
+// will be returned.
+func (st *State) GetAllUsersWithAuthInfo(ctx context.Context) ([]user.UserWithAuthInfo, error) {
 	db, err := st.DB()
 	if err != nil {
 		return nil, errors.Annotate(err, "getting DB access")
 	}
 
-	var usrs []user.User
+	var usrs []user.UserWithAuthInfo
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		getAllUsersQuery := `
-SELECT &User.*
+SELECT (user.uuid, user.name, user.display_name, user.created_by_uuid, user.created_at, user_authentication.last_login, user_authentication.disabled) AS (&User.*)
```

Comment:
This line is too long. What prevents selecting type.*?

If that can't be done, you can still:
- Break the columns into separate lines.
- Alias the table names so you don't have to repeat the long prefixes.

Reply:
done